src/app/InteractionModelEngine.cpp

```diff
@@ -408,6 +476,72 @@ Protocols::InteractionModel::Status InteractionModelEngine::OnReadInitialRequest
             });
         }
     }
+    else
+    {
+        System::PacketBufferTLVReader reader;
+
+        reader.Init(aPayload.Retain());
+
+        ReadRequestMessage::Parser readRequestParser;
+        VerifyOrReturnError(readRequestParser.Init(reader) == CHIP_NO_ERROR, Status::InvalidAction);
+
+        size_t requestedAttributePathCount = 0;
+        size_t requestedEventPathCount     = 0;
+
+        AttributePathIBs::Parser attributePathListParser;
+        bool hasValidAttributePath = false;
+
+        CHIP_ERROR err = readRequestParser.GetAttributeRequests(&attributePathListParser);
+        if (err == CHIP_NO_ERROR)
+        {
+            auto subjectDescriptor = apExchangeContext->GetSessionHandle()->AsSecureSession()->GetSubjectDescriptor();
+            err =
+                ParseAttributePaths(subjectDescriptor, attributePathListParser, hasValidAttributePath, requestedAttributePathCount);
+            if (err != CHIP_NO_ERROR)
+            {
+                return Status::InvalidAction;
+            }
+        }
+        else if (err != CHIP_ERROR_END_OF_TLV)
+        {
+            return Status::InvalidAction;
+        }
+
+        EventPathIBs::Parser eventpathListParser;
+        err = readRequestParser.GetEventRequests(&eventpathListParser);
+        if (err == CHIP_NO_ERROR)
+        {
+            TLV::TLVReader pathReader;
+            eventpathListParser.GetReader(&pathReader);
+            TLV::Utilities::Count(pathReader, requestedEventPathCount, false);
+        }
+        else if (err != CHIP_ERROR_END_OF_TLV)
+        {
+            return Status::InvalidAction;
+        }
+
+        if (requestedAttributePathCount == 0 && requestedEventPathCount == 0)
+        {
+            ChipLogError(InteractionModel,
+                         "Subscription from [%u:" ChipLogFormatX64 "] has zero path for attribute and event. Rejecting request.",
```

Comment:
This is not a subscription.  And note the wording suggestion for the subscription case.

Reply:
corrected, thanks